src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs

```diff
@@ -2476,6 +2468,359 @@ public static void DCJS_RoundtrippingDateTime()
+
+        var dt35832 = new DateTime(DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day, 3, 58, 32);
+        var dt = new DateTime(DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day);
+        var expected_DT_list = "[\"03:58:32.00 a.m.\",\"12:00:00.00 a.m.\",\"12:00:00.00 a.m.\",\"03:58:32.00 a.m.\"]";
+        dcjsSettings = new DataContractJsonSerializerSettings() { DateTimeFormat = jsonTypes.DTF_hmsFt };
+        var actual4 = SerializeAndDeserialize(jsonTypes.DT_List, expected_DT_list, dcjsSettings);
+        Assert.NotNull(actual4);
+        Assert.True(actual4[0] == dt35832);
+        Assert.True(actual4[1] == dt);
+        Assert.True(actual4[2] == dt);
+        Assert.True(actual4[3] == dt35832);
+
+        var expected_DT_dictionary = "[{\"Key\":\"03:58:32.00 a.m.\",\"Value\":\"03:58:32.00 a.m.\"},{\"Key\":\"12:00:00.00 a.m.\",\"Value\":\"12:00:00.00 a.m.\"}]";
+        dcjsSettings = new DataContractJsonSerializerSettings() { DateTimeFormat = jsonTypes.DTF_hmsFt };
+        var actual5 = SerializeAndDeserialize(jsonTypes.DT_Dictionary, expected_DT_dictionary, dcjsSettings);
+        Assert.NotNull(actual5);
+        Assert.True(actual5[dt35832] == dt35832);
+        Assert.True(actual5[dt] == dt);
+
+        var dateTime = new DateTime(2008, 5, 1, 8, 6, 32, DateTimeKind.Local);
+        string expectedOutput = dateTime.ToString("yyyy-MM-ddTHH:mm:ss.fffK", DateTimeFormatInfo.CurrentInfo);
+        expectedOutput = String.Format("\"{0}\"", expectedOutput);
+        dcjsSettings = new DataContractJsonSerializerSettings() { DateTimeFormat = jsonTypes.DTF_DefaultFormatProviderIsDateTimeFormatInfoDotCurrentInfo };
+        var actual6 = SerializeAndDeserialize(dateTime, expectedOutput, dcjsSettings);
+        Assert.NotNull(actual6);
+        Assert.True(actual6 == dateTime);
+    }
+
+    [Fact]
+    public static void DCJS_VerifyDateTimeForDateTimeFormat()
+    {
+        var jsonTypes = new JsonTypes();
+        var dateTime = new DateTime(DateTime.Now.Year, 12, 1);
+        var expectedString = "\"" + dateTime.ToString("MMMM") + "\"";
+        var dcjsSettings = new DataContractJsonSerializerSettings()
+        {
+            DateTimeFormat = jsonTypes.DTF_MMMM,
+            UseSimpleDictionaryFormat = true,
+            EmitTypeInformation = EmitTypeInformation.AsNeeded,
+            KnownTypes = new List<Type>()
+        };
+        var actual = SerializeAndDeserialize(dateTime, expectedString, dcjsSettings);
+        Assert.NotNull(actual);
+        Assert.True(actual == dateTime);
+
+        dcjsSettings = new DataContractJsonSerializerSettings()
+        {
+            DateTimeFormat = jsonTypes.DTF_hmsFt,
+            UseSimpleDictionaryFormat = true,
+            EmitTypeInformation = EmitTypeInformation.AsNeeded,
+            KnownTypes = new List<Type>()
+        };
+
+        var value = (Environment.OSVersion.Platform == PlatformID.Unix) ? "\"03:58:32.00 a. m.\"" : "\"03:58:32.00 a.m.\"";      
```

Comment:
>  [](start = 91, length = 1)

Interesting, I wondered why the behaviors differ between windows and Unix. @stone-li can you please